sympy/polys/matrices/domainmatrix.py

```diff
@@ -21,6 +21,37 @@
 
 
 class DomainMatrix:
+    r"""
+
+        The idea behind DomainMatrix is to associate :py:class:`~.Domain` with matrix,
+        which could be found analogous to numpy arrays with "dtype".
+        In the DomainMatrix, each matrix has a domain such as ZZ(Integers) or QQ<sqrt(2)>[Algebraic Number Field].
+
+        DomainMatrix uses "domain elements" for its internal representation which makes it more faster for many common operations
+        than current sympy Matrix class, but this advantage makes it not entirely compatible with Matrix.
+
+
+        Creating a DomainMatrix from the existing Matrix class:
+
+        >>> from sympy import Matrix
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> matrix = Matrix([[99, 16, 30, 62], [65, 79, 94, 15], [38, 13, 31, 47], [17, 2, 53, 78]])
+        >>> domainMatrix = DomainMatrix.from_Matrix(matrix)
+        >>> domainMatrix
+        DomainMatrix([[99, 16, 30, 62], [65, 79, 94, 15], [38, 13, 31, 47], [17, 2, 53, 78]], (4, 4), ZZ)
+
+
+        See Also
+        ========
+        ddm, Domain, Poly
+
+
+        References
+        ==========
+
+        ..[1] https://docs.sympy.org/dev/modules/polys/domainsintro.html
```

Comment:
Don't use an external link to a page in the docs. You should be able to use a sphinx reference to link to this.

Reply:
Will these work?
```
See Also
========

:py:class:`~.ddm`
:py:class:`~.Domain`

References
==========

.. module:: sympy.polys
.. module:: sympy.polys.domainsintro
```
Thanks.

Reply:
Sure, thanks.